### ISSUE TITLE
Stringify with {format: "RFC1738"}

### ISSUE THREAD

**Issue by JosuaMeier** [ljharb/qs#390]
I want to stringify a object with RFC1738 format.
As defined in RFC1738 "(" and ")" should be allowed and therefore not be encoded.
```
let urlQuery = {
  foo(ref): "BAR"
}
qs.stringify(urlQuery, { format: 'RFC1738' });
```

### Expected Result
`"foo(ref)=BAR"`

### Current Result
`"foo%28ref%29=BAR"`

**Comment by ljharb** [comment]
Thanks, seems like this is because we're using the same default `encoder` for every format.

**Comment by huangfushan** [comment]
![image](https://user-images.githubusercontent.com/23737468/132836691-1672274a-1d79-423c-baec-99dae54b850b.png)

Has this problem been solved? My version is 6.10.1, it seems that this problem still exists？
